lib/Resource.js

```diff
@@ -120,23 +120,24 @@ Resource.prototype = {
                     );
                 }
 
+                var errData = {
+                    code: res.statusCode,
```

Comment:
Can we call this `statusCode`? might be good to leave `code` for our own internal error codes.